File: dashboard/src/api/analytics.ts
import { useQuery } from '@tanstack/react-query';
import { apiFetch } from './client';

export interface ToolUsageItem {
  tool_name: string;
  count: number;
}

export interface AgentProductivityItem {
  agent_id: string;
  agent_name: string;
  activity_count: number;
  tools_used: number;
  last_active: string;
}

export interface TimelineItem {
  hour: string;
  count: number;
}

export interface TeamOverview {
  total_activities: number;
  total_agents: number;
  active_agents: number;
  tool_distribution: ToolUsageItem[];
  agent_productivity: AgentProductivityItem[];
}

interface ApiResponse<T> {
  success: boolean;
  data: T;
}

export function useToolUsage(teamId?: string) {
  return useQuery({
    queryKey: ['analytics', 'tool-usage', teamId],
    queryFn: async () => {
      const res = await apiFetch<ApiResponse<ToolUsageItem[]>>(
        `/api/analytics/tool-usage${teamId ? `?team_id=${teamId}` : ''}`,
      );
      return res.data;
    },
    refetchInterval: 30_000,
  });
}

export function useAgentProductivity(teamId?: string) {
  return useQuery({
    queryKey: ['analytics', 'agent-productivity', teamId],
    queryFn: async () => {
      const res = await apiFetch<ApiResponse<AgentProductivityItem[]>>(
        `/api/analytics/agent-productivity${teamId ? `?team_id=${teamId}` : ''}`,
      );
      return res.data;
    },
    refetchInterval: 30_000,
  });
}

export function useActivityTimeline(teamId?: string, hours = 24) {
  return useQuery({
    queryKey: ['analytics', 'timeline', teamId, hours],
    queryFn: async () => {
      const res = await apiFetch<ApiResponse<TimelineItem[]>>(
        `/api/analytics/timeline${teamId ? `?team_id=${teamId}&hours=${hours}` : `?hours=${hours}`}`,
      );
      return res.data;
    },
    refetchInterval: 60_000,
  });
}

export function useTeamOverview(teamId: string) {
  return useQuery({
    queryKey: ['analytics', 'team-overview', teamId],
    queryFn: async () => {
      const res = await apiFetch<ApiResponse<TeamOverview>>(
        `/api/analytics/team-overview?team_id=${teamId}`,
      );
      return res.data;
    },
    enabled: !!teamId,
    refetchInterval: 30_000,
  });
}

File: src/aiteam/api/routes/analytics.py
"""AI Team OS — 数据分析统计路由."""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Depends, Query

from aiteam.api.deps import get_repository
from aiteam.storage.repository import StorageRepository

router = APIRouter(prefix="/api/analytics", tags=["analytics"])


@router.get("/tool-usage")
async def get_tool_usage(
    team_id: str | None = Query(None),
    repo: StorageRepository = Depends(get_repository),
) -> dict[str, Any]:
    """工具使用分布统计."""
    data = await repo.count_activities_by_tool(team_id=team_id)
    return {"success": True, "data": data}


@router.get("/agent-productivity")
async def get_agent_productivity(
    team_id: str | None = Query(None),
    repo: StorageRepository = Depends(get_repository),
) -> dict[str, Any]:
    """Agent产能指标."""
    data = await repo.get_agent_productivity(team_id=team_id)
    return {"success": True, "data": data}


@router.get("/timeline")
async def get_activity_timeline(
    team_id: str | None = Query(None),
    hours: int = Query(24, ge=1, le=168),
    repo: StorageRepository = Depends(get_repository),
) -> dict[str, Any]:
    """活动时间线（按小时聚合）."""
    data = await repo.get_activity_timeline(team_id=team_id, hours=hours)
    return {"success": True, "data": data}


@router.get("/team-overview")
async def get_team_overview(
    team_id: str = Query(...),
    repo: StorageRepository = Depends(get_repository),
) -> dict[str, Any]:
    """团队整体统计概览."""
    # 工具分布
    tool_dist = await repo.count_activities_by_tool(team_id=team_id)

    # Agent产能
    productivity = await repo.get_agent_productivity(team_id=team_id)

    # 活跃Agent数
    agents = await repo.list_agents(team_id)
    active_agents = [a for a in agents if a.status.value == "busy"]

    # 总活动数
    total_activities = sum(p["activity_count"] for p in productivity)

    return {
        "success": True,
        "data": {
            "total_activities": total_activities,
            "total_agents": len(agents),
            "active_agents": len(active_agents),
            "tool_distribution": tool_dist,
            "agent_productivity": productivity,
        },
    }
